tests/v3_api/test_registry.py

```diff
@@ -0,0 +1,189 @@
+from common import *   # NOQA
+import pytest
+
+namespace = {"p_client": None, "ns": None, "cluster": None, "project": None}
+REGISTRY_USER_NAME = os.environ.get('REGISTRY_USER_NAME', "None")
+REGISTRY_PASSWORD = os.environ.get('REGISTRY_PASSWORD', "None")
+TEST_CLIENT_IMAGE = os.environ.get('TEST_CLIENT_IMAGE', "None")
+REGISTRY = os.environ.get('REGISTRY', "None")
+
+
+def test_create_registry_single_namespace():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+
+    create_registry_validate_workload(p_client, ns)
+
+
+def test_create_registry_all_namespace():
+    p_client = namespace["p_client"]
+    c_client = namespace["c_client"]
+    cluster = namespace["cluster"]
+    project = namespace["project"]
+
+    ns = namespace["ns"]
+    create_registry_validate_workload(p_client, ns, allns=True)
+
+    # Create and validate workload in a new namespace
+    new_ns = create_ns(c_client, cluster, project)
+    create_validate_workload(p_client, new_ns)
+
+
+def test_delete_registry_all_namespace():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+
+    registry = create_registry_validate_workload(p_client, ns, allns=True)
+    delete_registry(p_client, registry)
+
+
+def test_delete_registry_single_namespace():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+
+    registry = create_registry_validate_workload(p_client, ns)
+    delete_registry(p_client, registry)
+
+
+def test_edit_registry_single_namespace():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+
+    registry = create_registry_validate_workload(p_client, ns)
+    # Update registry with invalid password
+    new_registries = {REGISTRY: {"username": REGISTRY_USER_NAME,
+                      "password": "abcdef"}}
+    p_client.update(registry, name=registry.name,
+                    namespaceId=ns['name'],
+                    registries=new_registries)
+    # Validate workload after registry update
+    create_validate_workload_after_registry_update(p_client, ns)
```

Comment:
@soumyalj In case of edit registry , it is better to start with an invalid credential and make sure workload deployment fails. Then edit with correct credentials and make sure workload deployment succeeds.